### PR TITLE
[CLI-781-2]: Testing for the urgent network issue

### PR DESCRIPTION
**Pull request by KyleGoyette**
Description
-----------

Updates and adds tests using mocking for the previous urgent ticket. Fixes a flake issue.

Testing
-------
The secret is this PR is testing.

### CODE REVIEW

File: tests/utils/mock_server.py [hunk 1]
```diff
@@ -297,7 +297,6 @@ def graphql():
         if body["variables"].get("files"):
             requested_file = body["variables"]["files"][0]
             ctx["requested_file"] = requested_file
-            app.logger.info("graphql requested file: %s", requested_file)
```
Comment (KyleGoyette) [inline tests/utils/mock_server.py:300]: This introduced flakeyness to a parallel runs test.

Comment (vanpelt) [reply tests/utils/mock_server.py:300]: Wait, how was this related to "flakeyness"?

**Review by vanpelt — APPROVED** [review]

File: tests/utils/mock_server.py (hunk 1 above)
Comment (KyleGoyette) [reply tests/utils/mock_server.py:300]: I honestly don't know. But with that line I was seeing about a 10% error rate on `test_parallel_runs`.

File: tests/wandb_integration_test.py [hunk 2]
```diff
@@ -362,27 +357,48 @@ def test_version_retired(
...
-    fpath = os.path.join(run.dir, "testFile")
+    test_settings.update({"start_method": "thread"})
+
+    def mock_min_size(self, size):
+        return 5
+
+    mocker.patch("wandb.filesync.dir_watcher.PolicyLive.RATE_LIMIT_SECONDS", 5)
+    mocker.patch(
+        "wandb.filesync.dir_watcher.PolicyLive.min_wait_for_size", mock_min_size
+    )
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply tests/wandb_integration_test.py:369]: After some more tuning I was able to get it down to 2 seconds and still have it pass consistently.

**Pull request merged**